packages/drivers/odsp-driver/src/odspDriverUrlResolverForShareLink.ts

```diff
@@ -30,15 +30,17 @@ import { getFileLink } from "./getFileLink";
 export class OdspDriverUrlResolverForShareLink implements IUrlResolver {
     private readonly logger: ITelemetryLogger;
     private readonly sharingLinkCache = new PromiseCache<string, string>();
-    private readonly instrumentedTokenFetcher: TokenFetcher<OdspResourceTokenFetchOptions>;
+    private readonly instrumentedTokenFetcher: TokenFetcher<OdspResourceTokenFetchOptions> | undefined;
     public constructor(
-        tokenFetcher: TokenFetcher<OdspResourceTokenFetchOptions>,
+        tokenFetcher: TokenFetcher<OdspResourceTokenFetchOptions> | undefined = undefined,
```

Comment:
Would it be cleaner if it was required, but default value would be a function that throws?
Always looking for reduction of states :)

Reply:
I think this is the cleanest/easiest way to allow for mode where resolver is used without any expectation of share link being fetched and subsequently used. Token Fetcher is used as part of share link generation code path and it will be semantically awkward to distinguish between the case where token fetch error has to be logged and has to be skipped. I think that having fetcher undefined is stronger statement: I did not specify it on purpose so there is no need to log token fetch failure. 
I added JsDocs comments to provide documentation details